daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/ProcessingError.scala

```diff
@@ -47,7 +47,7 @@ abstract class ProcessingError protected (
    * looks like the same as other parse errors to tests that search for the
    * "Parse Error" string.
    */
-  def toParseError = new ParseError(schemaContext, dataContext, Maybe(this), maybeFormatString, args: _*)
+  def toParseError = new ParseError(schemaContext, dataContext, maybeCause, maybeFormatString, args: _*)
```

Comment:
Changing this to maybeCause isn't correct. 
'this' object itself becomes the cause of the new ParseError. You are jumping ahead one in the chain of nested exceptions by using maybeCause here.

Reply:
Updated PR.